### ISSUE TITLE
v0.5.1版本在最新的flutter-v1.7.8无法运行

### ISSUE THREAD

**Issue by chavesgu** [fluttercandies/extended_image#28]

**Comment by chavesgu** [comment]
Compiler message:
file:///Users/guchunwen/Documents/software/flutter/.pub-cache/hosted/pub.dartlang.org/extended_image-0.5.1/lib/src/extended_image.dart:733:59: Error: The argument type 'void Function(ImageInfo, bool)' can't be assigned to the parameter type 'ImageStreamListener'.
 - 'ImageInfo' is from 'package:flutter/src/painting/image_stream.dart' ('file:///Users/guchunwen/Documents/software/flutter/packages/flutter/lib/src/painting/image_stream.dart').
 - 'ImageStreamListener' is from 'package:flutter/src/painting/image_stream.dart' ('file:///Users/guchunwen/Documents/software/flutter/packages/flutter/lib/src/painting/image_stream.dart').
Try changing the type of the parameter, or casting the argument to 'ImageStreamListener'.
    if (_isListeningToStream) _imageStream.removeListener(_handleImageChanged);
                                                          ^
file:///Users/guchunwen/Documents/software/flutter/.pub-cache/hosted/pub.dartlang.org/extended_image-0.5.1/lib/src/extended_image.dart:744:53: Error: No named parameter with the name 'onError'.
      _imageStream.addListener(_handleImageChanged, onError: _loadFailed);
                                                    ^^^^^^^
file:///Users/guchunwen/Documents/software/flutter/.pub-cache/hosted/pub.dartlang.org/extended_image-0.5.1/lib/src/extended_image.dart:749:51: Error: No named parameter with the name 'onError'.
    _imageStream.addListener(_handleImageChanged, onError: _loadFailed);

**Comment by ecombharadwaj** [comment]
facing same issue 

Compiler message:
file:///sdk/flutter/.pub-cache/hosted/pub.dartlang.org/flushbar-1.5.3/lib/flushbar.dart:228:15: Error: The method 'detach' isn't defined for the class 'FocusScopeNode'.
 - 'FocusScopeNode' is from 'package:flutter/src/widgets/focus_manager.dart' ('file:///sdk/flutter/packages/flutter/lib/src/widgets/focus_manager.dart').
Try correcting the name to the name of an existing method, or defining a method named 'detach'.
    focusNode.detach();
              ^^^^^^
file:///sdk/flutter/.pub-cache/hosted/pub.dartlang.org/extended_image-0.5.1/lib/src/extended_image.dart:733:59: Error: The argument type 'void Function(ImageInfo, bool)' can't be assigned to the parameter type 'ImageStreamListener'.
 - 'ImageInfo' is from 'package:flutter/src/painting/image_stream.dart' ('file:///sdk/flutter/packages/flutter/lib/src/painting/image_stream.dart').
 - 'ImageStreamListener' is from 'package:flutter/src/painting/image_stream.dart' ('file:///sdk/flutter/packages/flutter/lib/src/painting/image_stream.dart').
Try changing the type of the parameter, or casting the argument to 'ImageStreamListener'.
    if (_isListeningToStream) _imageStream.removeListener(_handleImageChanged);
                                                          ^
file:///sdk/flutter/.pub-cache/hosted/pub.dartlang.org/extended_image-0.5.1/lib/src/extended_image.dart:744:53: Error: No named parameter with the name 'onError'.
      _imageStream.addListener(_handleImageChanged, onError: _loadFailed);
                                                    ^^^^^^^
file:///sdk/flutter/.pub-cache/hosted/pub.dartlang.org/extended_image-0.5.1/lib/src/extended_image.dart:749:51: Error: No named parameter with the name 'onError'.
    _imageStream.addListener(_handleImageChanged, onError: _loadFailed);
                                                  ^^^^^^^
file:///sdk/flutter/.pub-cache/hosted/pub.dartlang.org/extended_image-0.5.1/lib/src/extended_image.dart:755:33: Error: The argument type 'void Function(ImageInfo, bool)' can't be assigned to the parameter type 'ImageStreamListener'.
 - 'ImageInfo' is from 'package:flutter/src/painting/image_stream.dart' ('file:///sdk/flutter/packages/flutter/lib/src/painting/image_stream.dart').
 - 'ImageStreamListener' is from 'package:flutter/src/painting/image_stream.dart' ('file:///sdk/flutter/packages/flutter/lib/src/painting/image_stream.dart').
Try changing the type of the parameter, or casting the argument to 'ImageStreamListener'.
    _imageStream.removeListener(_handleImageChanged);
                                ^
file:///sdk/flutter/.pub-cache/hosted/pub.dartlang.org/photo_view-0.1.3/lib/photo_view.dart:237:24: Error: The argument type 'Null Function(ImageInfo, bool)' can't be assigned to the parameter type 'ImageStreamListener'.
 - 'ImageInfo' is from 'package:flutter/src/painting/image_stream.dart' ('file:///sdk/flutter/packages/flutter/lib/src/painting/image_stream.dart').
 - 'ImageStreamListener' is from 'package:flutter/src/painting/image_stream.dart' ('file:///sdk/flutter/packages/flutter/lib/src/painting/image_stream.dart').
Try changing the type of the parameter, or casting the argument to 'ImageStreamListener'.
    stream.addListener(listener);
                       ^
file:///sdk/flutter/.pub-cache/hosted/pub.dartlang.org/photo_view-0.1.3/lib/photo_view.dart:239:29: Error: The argument type 'Null Function(ImageInfo, bool)' can't be assigned to the parameter type 'ImageStreamListener'.
 - 'ImageInfo' is from 'package:flutter/src/painting/image_stream.dart' ('file:///sdk/flutter/packages/flutter/lib/src/painting/image_stream.dart').
 - 'ImageStreamListener' is from 'package:flutter/src/painting/image_stream.dart' ('file:///sdk/flutter/packages/flutter/lib/src/painting/image_stream.dart').
Try changing the type of the parameter, or casting the argument to 'ImageStreamListener'.
      stream.removeListener(listener);

**Comment by zmtzawqlp** [comment]
check https://github.com/fluttercandies/extended_image/issues/6

i push codes to pub later
